catroid/src/main/java/org/catrobat/catroid/devices/arduino/ArduinoListener.java

```diff
@@ -164,11 +155,28 @@ public int getAnalogPin5() {
 		return analogPin5;
 	}
 
-	public int getPortValue(int pin) {
-		return portValue[pin];
+	public int getPinValue(int pin) {
+		if ((pin >= 0) && (pin < ArduinoImpl.NUMBER_OF_DIGITAL_PINS)) {
+			int port = (pin + 7) / 8;
+			int index = pin % 8;
+			return ArduinoImpl.getBit(this.portValue[port], index);
+		}
+		return 0;
+	}
+
+	public void setPinValue(int pin, int value) {
+		if ((pin >= 0) && (pin < ArduinoImpl.NUMBER_OF_DIGITAL_PINS)) {
+			int port = (pin + 7) / 8;
+			int index = pin % 8;
+			this.portValue[port] = ArduinoImpl.setBit(this.portValue[port], index, value);
+		}
```

Comment:
`getPinValue` and `setPinValue` share a lot of code.
I would suggest you create 3 methods. A method...
* for the guard clause if the pin is in range.
* to get the port. You can reuse it to create the size of the port array and if still needed, in `ArduinoImpl.setDigitalArduinoPin`
* to get the pin on the port. Can be reused in `ArduinoImpl.setDigitalArduinoPin` as well, if still needed.